### PR TITLE
Add support for CONSTANT_Dynamic

### PR DESCRIPTION
**Pull request by gtoison**
Class files modified by jacoco cause the spotbugs analysis to crash due to #2759

### CODE REVIEW

**Comment by gtoison** [comment]
```
Issue2759Test > testIssue(SpotBugsRunner) FAILED
    java.lang.AssertionError: Analysis failed with exception. Check stderr for detail.
        at edu.umd.cs.findbugs.test.AnalysisRunner.run(AnalysisRunner.java:121)
        at edu.umd.cs.findbugs.test.SpotBugsRunner.performAnalysis(SpotBugsRunner.java:61)
        at edu.umd.cs.findbugs.test.SpotBugsRunner.performAnalysis(SpotBugsRunner.java:75)
        at edu.umd.cs.findbugs.detect.Issue2759Test.testIssue(Issue2759Test.java:23)
```

It would  be great if the build could upload the test results in case there are failures

**Comment by JuditKnoll** [comment]
There was a PR about `CONSTANT_Dynamic` partial support, which may be connected, but it looks like the author decided to delete the spotbugs fork. Fortunately, I took a look at it locally earlier, so I could push it under my fork: https://github.com/JuditKnoll/spotbugs/tree/support-constant-dynamic

**Comment by gtoison** [comment]
Thanks @JuditKnoll that's very helpful!

**Review by ThrawnCA — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -28,6 +28,9 @@ subprojects {
 allprojects {
   repositories {
     mavenCentral()
+    maven {
```
Comment (gtoison) [inline build.gradle:31]: This seems to be working, the only missing piece is that we're using a BCEL release candidate.
Hopefully they will release the new version soon, then I'll remove this bit

Comment (hazendaz) [reply build.gradle:31]: @gtoison Its ready to be removed now. I've merged in bcel 6.8.1

**Review by JuditKnoll — APPROVED** [review]

**Pull request merged**